mlflow/diviner.py

```diff
@@ -192,7 +214,71 @@ def save_model(
+    for the training data submitted to `fit`, directly save the Diviner model object.
+    If the Diviner model was fit by using a Spark DataFrame, save the model components separately.
+    The metadata and ancillary files to write (JSON and Pandas DataFrames) are written directly
+    to a fuse mount location, which the Spark DataFrame that contains the individual serialized
+    Diviner model objects is written by using the 'dbfs:' scheme path that Spark recognizes.
+    """
+    fit_with_spark = False
+
+    save_path = str(path.joinpath(_MODEL_BINARY_FILE_NAME))
+
+    if hasattr(diviner_model, "_fit_with_spark") and diviner_model._fit_with_spark:
+        # Validate that the path is a relative path early in order to fail fast prior to attempting
+        # to write the (large) DataFrame to a tmp DFS path first and raise a path validation
+        # Exception within MLflow when attempting to copy the temporary write files from DFS to
+        # the file system path provided.
+        if not os.path.isabs(path):
+            raise MlflowException(
+                "The save path provided must be a run-relative path. "
+                f"The path submitted, '{path}' is an absolute path."
+            )
+
+        # Create a temporary DFS location to write the Spark DataFrame containing the models to.
+        tmp_path = generate_tmp_dfs_path(kwargs.get("dfs_tmpdir", MLFLOW_DFS_TMP.get()))
+
+        # Save the model Spark DataFrame to the temporary DFS location
+        diviner_model._save_model_df_to_path(tmp_path, **kwargs)
+
+        diviner_data_path = os.path.abspath(save_path)
+
+        tmp_fuse_path = dbfs_hdfs_uri_to_fuse_path(tmp_path)
+        shutil.move(src=tmp_fuse_path, dst=diviner_data_path)
+
+        # Save the model metadata to the path location
+        diviner_model._save_model_metadata_components_to_path(path=diviner_data_path)
+        fit_with_spark = True
+    else:
+        diviner_model.save(save_path)
+    return fit_with_spark
+
+
+def _load_model_fit_in_spark(local_model_path: str, flavor_conf, **kwargs):
+    """
+    Loads a Diviner model that has been fit (and saved) in the Spark variant.
+    """
+    # NB: To load the model DataFrame (which is a Spark DataFrame), Spark requires that the file
+    # partitions are in DFS. In order to facilitate this, the model DataFrame (saved as parquet)
+    # will be copied to a temporary DFS location. The remaining files can be read directly from
+    # the local file system path, which is handled within the Diviner APIs.
+    import diviner
+
+    dfs_temp_directory = generate_tmp_dfs_path(kwargs.get("dfs_tmpdir", MLFLOW_DFS_TMP.get()))
+    dfs_fuse_directory = dbfs_hdfs_uri_to_fuse_path(dfs_temp_directory)
+    os.makedirs(dfs_fuse_directory)
+    shutil_copytree_without_file_permissions(src_dir=local_model_path, dst_dir=dfs_fuse_directory)
+
+    diviner_instance = getattr(diviner, flavor_conf[_MODEL_TYPE_KEY])
```

Comment:
Does `diviner` always have `flavor_conf[_MODEL_TYPE_KEY]`?

Reply:
yep. We save the `model_type` as part of writing out the `MLmodel` file; it's the class name of (currently) either `GroupedProphet` or `GroupedPmdarima`. If we create additional variants in Diviner in the future, those will be options for this attribute as well.